mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/ClusterID.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.mantisrx.server.master.resourcecluster;
+
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+
+/**
+ * Represents the cluster that the task executor belongs to.
+ */
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+@Value
+public class ClusterID {
+  String resourceID;
+
+  public static ClusterID of(String clusterIdStr) {
```

Comment:
I really love the way static factory methods named `of` read in client code. `ClusterId.of("m5.2xl-1")` reads so cleanly to a user even compared to the builder `ClusterId.builder().name("m5.2xl-1").build()`. However I think this can be achieved with Lombok using `@RequiredArgsConstructor(staticName="of")`.

Reply:
Very true. "of" is definitely cleaner on the single arg ctor case.

Reply:
That is actually a good point, it can get confusing on the multi-arg constructor case. Especially if parameters are of the same type, which string represents what?

Reply:
Thanks for the suggestion. I have used the Lombok annotation in both the places where it makes sense.